src/main/java/com/gulteking/mqttbackendserver/serviceimpl/DeviceServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+package com.gulteking.mqttbackendserver.serviceimpl;
+
+import com.google.gson.Gson;
+import com.gulteking.mqttbackendserver.entity.Device;
+import com.gulteking.mqttbackendserver.model.DeviceResponse;
+import com.gulteking.mqttbackendserver.model.JsonResponse;
+import com.gulteking.mqttbackendserver.model.MqttTopicSubscribe;
+import com.gulteking.mqttbackendserver.repository.DeviceRepository;
+import com.gulteking.mqttbackendserver.service.DeviceService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class DeviceServiceImpl implements DeviceService {
+    @Autowired
+    DeviceRepository deviceRepository;
+
+    @Autowired
+    MqttSubscriberImpl mqttSubscriber;
+
+    @Autowired
+    MqttPublisherImpl mqttPublisher;
+
+    @Override
```

Comment:
add @Transactional where ever we are doing multiple write operations.

Reply:
we are not writing anything into database .

Reply:
ok